spec/factories/miq_server.rb

```diff
@@ -1,18 +1,34 @@
 FactoryGirl.define do
   factory :miq_server do
+    transient do
+      # reference to the miq_server (typically the master) that knows the zone
```

Comment:
Well, aside from the fact that there is only 1 master in a zone, but that can be controlled by the specs themselves.